### PR TITLE
the panic's param may be `$ty` but not `$tt`

### PR DESCRIPTION
**Pull request by H-ZeX**
Signed-off-by: H-ZeX <hzx20112012@gmail.com>

## What have you changed? (mandatory)

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- in `coprocessors::codec::data_type::scalars::impl_as_ref::ScalarValue::$name`, the `stringify!($tt)`  may be changed to `stringify!($ty)`   

## What are the type of the changes? (mandatory)

- Bug fix (change which fixes an issue)

## How has this PR been tested? (mandatory)
- test using this demo
   ```rust
   macro_rules! my_macro_2 {
       ($ty: tt) => {
           println!("{}", stringify!($ty));
           println!("{}", stringify!($tt));
       };
   }
   ``` 
- use `rustc -Z unstable-options --pretty expanded scalar.rs > /tmp/3.rs` and analysis the expanded code

## Does this PR affect documentation (docs) or release note? (mandatory)
no 

## Does this PR affect tidb-ansible update? (mandatory)
no

### CODE REVIEW

File: src/coprocessor/codec/data_type/scalar.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ macro_rules! impl_as_ref {
                     other => panic!(
                         "Cannot cast {} scalar value into {}",
                         other.eval_type(),
-                        stringify!($tt),
```
Comment (BusyJay) [inline src/coprocessor/codec/data_type/scalar.rs]: Hmm, why did it compile?

Comment (kennytm) [reply src/coprocessor/codec/data_type/scalar.rs]: @BusyJay it stringified to the string `"$ tt"` 🤣

**Comment by breeswish** [comment]
Nice catch. How did you find it?

**Comment by H-ZeX** [comment]
> Nice catch. How did you find it?

I found it when read the code, no special way : )

**Comment by kennytm** [comment]
/test

Why it doesn't automatically run now?

**Comment by kennytm** [comment]
/test

Previous spurious failure in https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/4094/pipeline/110

```
test cases::test_conf_change::test_tick_after_destroy ... test cases::test_conf_change::test_tick_after_destroy has been running for over 60 seconds

test cases::test_conf_change::test_write_after_destroy ... test cases::test_conf_change::test_write_after_destroy has been running for over 60 seconds

Sending interrupt signal to process
```

**Pull request merged**